tests/test_it.py

```diff
@@ -140,6 +148,16 @@ def test_C402_fail_2():
     ]
 
 
+def test_C402_fail_3():
+    errors = run_flake8("""
+        lst = [('a', 1), ('b', 2), ('c', 3)]
+        dict(pair for pair in lst if pair[1] % 2 == 0)
```

Comment:
This can be rewritten as:

```python
lst = [('a', 1), ('b', 2), ('c', 3)]
{k, v for k, v in lst if v % 2 == 0}
```

So, I think the flake8 error is beneficial. If this code went form "fail" to "pass", I'd consider that a false negative. I wrote this test specifically to avoid that.